From 4223991edd54d844e711be1e8e6281185f929489 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Tue, 15 Apr 2014 16:38:56 +0100
Subject: [PATCH 195/351] regulator: arizona-micsupp: Sync with upstream
 comments

Use of IS_ENABLED is preferred for configuration specific code, so use
that instead of of ifdef.

Change-Id: Iae2f7069e7781f3198a58f72392659f791d30eb7
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/regulator/arizona-micsupp.c |   30 +++++++-----------------------
 1 file changed, 7 insertions(+), 23 deletions(-)

diff --git a/drivers/regulator/arizona-micsupp.c b/drivers/regulator/arizona-micsupp.c
index 5ce6483..cd326af 100644
--- a/drivers/regulator/arizona-micsupp.c
+++ b/drivers/regulator/arizona-micsupp.c
@@ -196,7 +196,6 @@ static const struct regulator_init_data arizona_micsupp_ext_default = {
 	.num_consumer_supplies = 1,
 };
 
-#ifdef CONFIG_OF
 static int arizona_micsupp_of_get_pdata(struct arizona *arizona,
 					struct regulator_config *config)
 {
@@ -223,22 +222,6 @@ static int arizona_micsupp_of_get_pdata(struct arizona *arizona,
 	return 0;
 }
 
-static void arizona_micsupp_of_put_pdata(struct regulator_config *config)
-{
-	of_node_put(config->of_node);
-}
-#else
-static inline int arizona_micsupp_of_get_pdata(struct arizona *arizona,
-					       struct regulator_config *config)
-{
-	return 0;
-}
-
-static inline void arizona_micsupp_of_put_pdata(struct regulator_config *config)
-{
-}
-#endif
-
 static int arizona_micsupp_probe(struct platform_device *pdev)
 {
 	struct arizona *arizona = dev_get_drvdata(pdev->dev.parent);
@@ -281,10 +264,12 @@ static int arizona_micsupp_probe(struct platform_device *pdev)
 	config.driver_data = micsupp;
 	config.regmap = arizona->regmap;
 
-	if (!dev_get_platdata(arizona->dev)) {
-		ret = arizona_micsupp_of_get_pdata(arizona, &config);
-		if (ret < 0)
-			return ret;
+	if (IS_ENABLED(CONFIG_OF)) {
+		if (!dev_get_platdata(arizona->dev)) {
+			ret = arizona_micsupp_of_get_pdata(arizona, &config);
+			if (ret < 0)
+				return ret;
+		}
 	}
 
 	if (arizona->pdata.micvdd)
@@ -306,8 +291,7 @@ static int arizona_micsupp_probe(struct platform_device *pdev)
 		return ret;
 	}
 
-	if (!dev_get_platdata(arizona->dev))
-		arizona_micsupp_of_put_pdata(&config);
+	of_node_put(config.of_node);
 
 	platform_set_drvdata(pdev, micsupp);
 
-- 
1.7.9.5

